models/m_payment.erl

```diff
@@ -325,6 +373,7 @@ set_payment_status(PaymentId, Status, StatusDate, Context) ->
                         set status = $1,
                             status_date = $3
                         where id = $2
+                          and status <> $1
```

Comment:
I guess this is to "debounce" the operation; it skips if an other, parallel request for setting the same status "won" the race. In principle this should almost never happen as status == old status should be covered in the case above.